src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/RazorDiagnosticsEndpoint.cs

```diff
@@ -389,6 +388,10 @@ static bool IsInvalidNestingFromBody(OmniSharpVSDiagnostic d, SourceText sourceT
             }
 
             var owner = syntaxTree.GetOwner(sourceText, d.Range.End, logger);
+            if (owner is null)
```

Comment:
Oh interesting I wonder if this is also due to the O# race issue where we occasionally get wrong text / doc state. If it's not that then it'd be a compiler bug where there's a hole in the document

Reply:
I'm pretty sure it's the race. It only happens when I slam the LS with undo requests. So that DOES make this a little bit of a patch, but I think not throwing in the meantime is good.

I figure we should make GetOwner throw if it would be null once that race is fixed in order to find compiler bugs more easily, but that's a different discussion I think.